gwtp-core/gwtp-dispatch-rest/src/main/java/com/gwtplatform/dispatch/rest/emul/javax/ws/rs/core/Cookie.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright 2015 ArcBees Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package javax.ws.rs.core;
+
+public class Cookie {
+    public static final int DEFAULT_VERSION = 1;
+
+    protected static final String COOKIE_PARTS_PATTERN = "[;,]";
+
+    private final String name;
+    private final String value;
+    private final int version;
+    private final String path;
+    private final String domain;
+
+    public Cookie(
+            String name,
+            String value,
+            String path,
+            String domain,
+            int version) {
+        if (name == null) {
+            throw new IllegalArgumentException("name==null");
+        }
+
+        this.name = name;
+        this.value = value;
```

Comment:
There's nothing requiring a cookie value or name to be encoded. I agree it must be encoded, but how to do it is up to the user (base64, url encode, etc). For example, RestEasy won't encode/decode Cookie. GWT's Cookies does URLEncode the value but offers a way to disable that behavior.
http://stackoverflow.com/a/1969339/1627870